src/xcode/ENA/ENA/Source/Scenes/FileScanner/FileScannerCoordinatorViewModel.swift

```diff
@@ -59,6 +59,7 @@ protocol FileScannerProcessing {
 	var processingStarted: (() -> Void)? { get set }
 	var processingFinished: ((QRCodeResult) -> Void)? { get set }
 	var processingFailed: ((FileScannerError?) -> Void)? { get set }
+	var parsingFailed: ((QRCodeParserError) -> Void)? { get set }
```

Comment:
I would propose to keep the API clean. processingStarted -> processingFinished or processingFailed. You could add the QRCodeParserError as an associated value to FileScannerError and call processingFailed with this error.